nestjs/src/courses/courseTasks/course-tasks.service.ts

```diff
@@ -62,4 +71,27 @@ export class CourseTasksService {
       relations: ['task'],
     });
   }
+
+  public getTasksPendingDeadline(
+    courseId: number,
+    { deadlineWithinHours = 24 }: { deadlineWithinHours?: number } = {},
+  ) {
+    const now = new Date().toISOString();
+    const endDate = new Date();
+    endDate.setHours(endDate.getHours() + deadlineWithinHours);
```

Comment:
i would probably suggest to bring some library for date management. If less experience developer will start contribute, it could be very hard to avoid mistakes etc.. :)